redis/src/test/scala/zio/redis/ApiSpec.scala

```diff
@@ -39,10 +44,170 @@ object ApiSpec extends BaseSpec {
+            renamed <- rename(key, newKey).either
+          } yield assert(renamed)(isLeft)
+        },
+        testM("rename key to newkey if newkey does not yet exist") {
+          for {
+            key     <- uuid
+            newKey  <- uuid
+            value   <- uuid
+            _       <- set(key, value, None, None, None)
+            renamed <- renameNx(key, newKey)
+            v       <- get(newKey)
+          } yield assert(renamed)(isTrue) && assert(v)(isSome(equalTo(value)))
+        },
+        testM("try to rename non-existing key with RENAMENX command") {
+          for {
+            key     <- uuid
+            newKey  <- uuid
+            renamed <- renameNx(key, newKey).either
+          } yield assert(renamed)(isLeft)
+        },
+        testM("unlink existing key") {
+          for {
+            key     <- uuid
+            value   <- uuid
+            _       <- set(key, value, None, None, None)
+            removed <- unlink(key, Nil)
+          } yield assert(removed)(equalTo(1L))
+        },
+        testM("touch two existing keys") {
+          for {
+            key1    <- uuid
+            value1  <- uuid
+            _       <- set(key1, value1, None, None, None)
+            key2    <- uuid
+            value2  <- uuid
+            _       <- set(key2, value2, None, None, None)
+            touched <- touch(key1, List(key2))
+          } yield assert(touched)(equalTo(2L))
+        },
+        testM("set followed by scan") {
+          for {
+            key             <- uuid
+            value           <- uuid
+            _               <- set(key, value, None, None, None)
+            scan            <- scan(0L, None, None, None)
+            (next, elements) = scan
+          } yield assert(next)(isNonEmptyString) && assert(elements)(isNonEmpty)
+        },
+        testM("fetch random key") {
+          for {
+            key       <- uuid
+            value     <- uuid
+            _         <- set(key, value, None, None, None)
+            allKeys   <- keys("*")
+            randomKey <- randomKey()
+          } yield assert(allKeys)(contains(randomKey.get))
+        },
+        testM("check value type by key") {
+          for {
+            key1       <- uuid
+            value1     <- uuid
+            _          <- set(key1, value1, None, None, None)
+            key2       <- uuid
+            value2     <- uuid
+            _          <- lPush(key2)(value2)
+            stringType <- typeOf(key1)
+            listType   <- typeOf(key2)
+          } yield assert(stringType)(equalTo("string")) && assert(listType)(equalTo("list"))
+        },
+        testM("dump followed by restore") {
+          for {
+            key      <- uuid
+            value    <- uuid
+            _        <- set(key, value, None, None, None)
+            dumped   <- dump(key)
+            _        <- del(key, Nil)
+            restore  <- restore((key, 0L, dumped, None, None, None, None)).either
```

Comment:
Please define `Arg7` syntax class in `RedisCommand`, no need to pass tuple explicitly.